addons/hr_skills_survey/tests/test_certification_flow.py

```diff
@@ -0,0 +1,101 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from dateutil.relativedelta import relativedelta
+from freezegun import freeze_time
+
+from odoo import Command, fields
+from odoo.addons.survey.tests import common
+from odoo.tests import tagged
+from odoo.tests.common import HttpCase
+
+
+@tagged('-at_install', 'post_install', 'functional')
+class TestCertificationFlow(common.TestSurveyCommon, HttpCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.employee_emp = cls.env['hr.employee'].create({
+            'user_id': cls.user_emp.id,
+        })
+        cls.certification = cls.env['survey.survey'].create({
+            'access_mode': 'public',
+            'certification': True,
+            'certification_validity_months': 3,
+            'description': '<p>Description</p>',
+            'questions_layout': 'page_per_question',
+            'scoring_success_min': 85.0,
+            'scoring_type': 'scoring_without_answers',
+            'title': 'Test Certification',
+            'users_login_required': True,
+        })
+        cls.certification_q0 = cls._add_question(
+            cls, None, 'Test question ...', 'simple_choice',
+            sequence=2,
+            constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=cls.certification.id,
+            labels=[
+                {'value': 'Correct', 'is_correct': True, 'answer_score': 1.0},
+                {'value': 'Incorrect'},
+            ])
+
+    @freeze_time('2024-03-20')
+    def test_resume_line_creation(self):
+        """ Check that the resume line is correctly created upon certification completion.
+
+        As we test the method "survey.user_input._mark_done" which is called in sudo from
```

Comment:
Answering here as code has changed.

It's fine to run as admin then, a more complete test would call the appropriate route as a regular user but it will add a lot of noise in the code, let's keep it with the comment.